### PR TITLE
fix(admin): replicate empty site groups

### PR DESCRIPTION
**Pull request by cxymds**
## Type of Change
- [x] Bug Fix
- [ ] New Feature
- [ ] Documentation
- [ ] Performance Improvement
- [ ] Test/CI
- [ ] Refactor
- [ ] Other:

## Related Issues
N/A

## Summary of Changes
- Fix site-replication group handling so non-delete `group-info` updates always upsert the target group.
- Preserve empty-group replication by removing the special-case path that only updated status for `members=[]`.
- Add a regression test covering empty member group replication semantics.

## Checklist
- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines
- [ ] Passed `make pre-commit`
- [x] Added/updated necessary tests
- [ ] Documentation updated (if needed)
- [ ] CI/CD passed (if applicable)

## Impact
- [ ] Breaking change (compatibility)
- [ ] Requires doc/config/deployment update
- [x] Other impact:
Fixes site replication for empty IAM groups.

## Additional Notes
Verification run:
- `cargo test -p rustfs test_group_info_with_empty_members_still_requires_group_upsert`

`make pre-commit` was attempted before PR creation, but the workspace currently has unrelated baseline test failures in `rustfs` outside this change, so this PR is opened for review with that known limitation.

---

Thank you for your contribution! Please ensure your PR follows the community standards ([CODE_OF_CONDUCT.md](CODE_OF_CONDUCT.md)). If this is your first contribution, review the [CLA document](https://github.com/rustfs/cla/blob/main/cla/v1.md) and sign it by commenting `I have read and agree to the CLA.` on the PR.

### CODE REVIEW

File: rustfs/src/admin/handlers/site_replication.rs [hunk 1]
```diff
@@ -2973,4 +2969,16 @@ mod tests {
...
+    #[test]
+    fn test_group_info_with_empty_members_still_requires_group_upsert() {
+        let update = rustfs_madmin::GroupAddRemove {
+            group: "empty-group".to_string(),
+            members: vec![],
+            status: GroupStatus::Enabled,
+            is_remove: false,
+        };
+
+        assert!(group_info_requires_upsert(&update));
```
[the remark this answers is not in the document]
Comment (cxymds) [reply rustfs/src/admin/handlers/site_replication.rs:2982]: Refactored the group-info branch to use a shared apply-mode helper and replaced the predicate-only test with a behavior test that verifies an empty-member update creates the group and allows status application. Verification: `cargo test -p rustfs test_group_info_with_empty_members_upserts_group_and_sets_status`, `cargo fmt --all --check`.

**Review by GatewayJ — APPROVED** [review]
